### DISCUSSION TITLE
pydantic model for endpoint args?

### DISCUSSION THREAD

**Discussion by jamesbraza** [encode/starlette#2333] [Q&A] [answered]
I am looking to migrate to Starlette from FastAPI.  Currently, my endpoints revolve around `pydantic` models as the direct function arguments.

I am wondering how to do the migration, does Starlette support pydantic model schemas?

Is the intended workflow to go from Starlette `request` to pydantic model instantiation within the endpoint?

---

Aside: I see these other third party packages seems to make pydantic first class:

- https://www.starlette.io/third-party-packages/#ellar
- https://www.starlette.io/third-party-packages/#esmerald

However, I am looking for non-third party, just curious about Starlette itself

**Comment by Kludex** [comment, marked as answer]
> I am wondering how to do the migration, does Starlette support pydantic model schemas?

No.

> Is the intended workflow to go from Starlette request to pydantic model instantiation within the endpoint?

On FastAPI, you should define the body you want in the signature function.
